src/shared_modules/utils/rocksDBQueueCF.hpp

```diff
@@ -226,19 +258,41 @@ class RocksDBQueueCF final
         }
     }
 
-    U front(std::string_view id) const
+    U front(std::string_view id)
     {
         U value;
+
         if (const auto it {m_queueMetadata.find(id.data())}; it != m_queueMetadata.end())
         {
-            if (!m_db->Get(rocksdb::ReadOptions(),
-                           m_db->DefaultColumnFamily(),
-                           std::string(id) + "_" + std::to_string(it->second.head),
-                           &value)
-                     .ok())
+            // If the queue is empty, return an empty value.
+            if (it->second.size == 0)
             {
-                throw std::runtime_error("Failed to get front element, id: " + std::string {id} +
-                                         " key: " + std::to_string(it->second.head));
+                throw std::runtime_error("Failed to get front element, queue is empty");
+            }
+
+            // If the queue have bumps between elements, get the first element in increasing order.
+            auto match = false;
+            auto index = it->second.head;
+
+            while (!match)
+            {
+                if (const auto status = m_db->Get(rocksdb::ReadOptions(),
+                                                  m_db->DefaultColumnFamily(),
+                                                  std::string(id) + "_" + std::to_string(index),
+                                                  &value);
+                    status.ok())
+                {
+                    match = true;
+                    break;
+                }
+                else
+                {
+                    if (status != rocksdb::Status::NotFound())
+                    {
+                        throw std::runtime_error("Failed to get elements, error: " + status.code());
+                    }
```

Comment:
This can be part of an **else if**
```suggestion
                else if (status != rocksdb::Status::NotFound())
                {
                    throw std::runtime_error("Failed to get elements, error: " + status.code());
```

Reply:
I prefer if-else or if-elseif-else or if